File: 3213-count-subarrays-where-max-element-appears-at-least-k-times/count-subarrays-where-max-element-appears-at-least-k-times.js
var countSubarrays = function(nums, k) {
    let maxNum = Math.max(...nums)  
    let countOfMaxNumber = 0
    let output = 0
    let i = 0
    let j = 0

    while (j < nums.length) {
        if (nums[j] == maxNum) {
            countOfMaxNumber += 1
        }

        while (countOfMaxNumber >= k) {
            output += nums.length - j
            if (nums[i] == maxNum) countOfMaxNumber -= 1
            i += 1
        }

        j += 1
    }
    return output

};

File: 3213-count-subarrays-where-max-element-appears-at-least-k-times/count-subarrays-where-max-element-appears-at-least-k-times.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        i = 0
        j = 0
        output = 0
        max_elem = max(nums)
        count = 0
        
        while (j < len(nums)):
            if (nums[j] == max_elem):
                count += 1

            while (count >= k):
                output += len(nums) - j
                if (nums[i] == max_elem):
                    count -= 1
                i += 1

            j += 1

        return output

        